### PR TITLE
Added slider

### PR DESCRIPTION
**Pull request by Wasileuss**
Added a slider. Check it out @KratosHome

### CODE REVIEW

File: src/components/client/slider/slider.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            <Slide />
+          </SwiperSlide>
+          <SwiperSlide>
+            <Slide />
+          </SwiperSlide>
+          <SwiperSlide>
+            <Slide />
+          </SwiperSlide>
+        </Swiper>
+        <div className="mx-auto flex max-w-fit items-center justify-center gap-8">
```
Comment (KratosHome) [inline src/components/client/slider/slider.tsx:63]: rewrite on tailwind

File: src/components/client/slider/slider.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          <div className="custom-prev swiper-button-prev">
+            <Image
+              src="/icons/slide-prev.svg"
+              width={50}
+              height={60}
+              alt="prev"
+            />
+          </div>
+          <div className="custom-pagination swiper-pagination flex items-center justify-center gap-1"></div>
```
Comment (KratosHome) [inline src/components/client/slider/slider.tsx:72]: rewrite on tailwind

File: src/components/client/hero/hero.tsx [hunk 3]
```diff
@@ -148,6 +149,7 @@ export const Hero = () => {
           alt="slider"
         />
       </div>
+      <Slider />
```
Comment (KratosHome) [inline src/components/client/hero/hero.tsx:152]: 1. this should not be in hero, add in page
2. slider should be made reusable
props: title, subTitle, array with product 
3. example arr product arr: IProduct.d.ts

File: src/components/client/slider/slider.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            },
+            768: {
+              slidesPerView: 3,
+            },
+          }}
+          id="swiper"
+          className="custom-swiper mb-24 pb-16"
+        >
+          <SwiperSlide>
+            <Slide />
```
Comment (KratosHome) [inline src/components/client/slider/slider.tsx:48]: name is better: product

File: src/components/client/slider/slide.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import Image from 'next/image'
+
+export default function Slide() {
+  return (
+    <div className="max-w-88 relative flex min-h-fit w-full flex-col justify-between gap-[50px]">
```
Comment (KratosHome) [inline src/components/client/slider/slide.tsx:5]: add hover effect

File: src/components/client/slider/slide.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        <div className="flex items-end justify-between">
+          <div>
+            <p className="text-xs font-semibold text-[#7D7D7D] line-through sm:text-sm md:text-base lg:text-lg xl:text-lg">
+              130 грн
+            </p>
+            <p className="text-sm font-semibold sm:text-base lg:text-lg xl:text-xl">
+              110 грн
+            </p>
+          </div>
+          <button
```
Comment (KratosHome) [inline src/components/client/slider/slide.tsx:44]: - use button with shadcn
- move the button to the ui folder
- add a better effect, better than the standard one

File: src/components/client/slider/slider.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import 'swiper/css'
+import 'swiper/css/navigation'
+import 'swiper/css/pagination'
+import { Autoplay, Navigation, Pagination } from 'swiper/modules'
+import Image from 'next/image'
+import Slide from './slide'
+const Slider: React.FC = () => {
+  return (
+    <>
+      <div className="mx-auto mb-12 mt-10 max-w-6xl">
```
Comment (KratosHome) [inline src/components/client/slider/slider.tsx:12]: hero section have bead adapted to mobile, delete hero and adaptive this component to 375

File: package.json [hunk 8]
```diff
@@ -36,6 +36,7 @@
     "react-dom": "^18",
     "react-hook-form": "^7.53.1",
     "sass": "^1.80.4",
+    "swiper": "^11.1.15",
```
Comment (KratosHome) [inline package.json:39]: we can take swiper but splidejs is better, various animations and splidejs is not Russian like swiper

**Review by KratosHome — APPROVED** [review]

File: src/components/client/slider/slider.tsx (hunk 2 above)
Comment (Wasileuss) [reply src/components/client/slider/slider.tsx:72]: Not actual. Swiperjs changed to splidejs

File: src/components/client/hero/hero.tsx (hunk 3 above)
Comment (Wasileuss) [reply src/components/client/hero/hero.tsx:152]: Done

File: src/components/client/slider/slider.tsx (hunk 4 above)
Comment (Wasileuss) [reply src/components/client/slider/slider.tsx:48]: Done

File: package.json (hunk 8 above)
Comment (Wasileuss) [reply package.json:39]: I've changed it, but I still need to customise the arrows and pagination

Comment (Wasileuss) [reply package.json:39]: Done

File: src/components/client/slider/slider.tsx (hunk 7 above)
Comment (Wasileuss) [reply src/components/client/slider/slider.tsx:12]: Adapted without hero section

File: src/components/client/slider/product.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import Button from '@/components/client/ui/button'
+
+type ProductProps = {
+  product: IProduct
+}
+
+const Product = ({ product }: ProductProps) => {
+  return (
+    <div className="max-w-88 relative flex h-full w-full flex-col justify-between gap-[50px] p-2 transition-transform hover:scale-105 hover:shadow-2xl">
+      {/* Badges */}
```
Comment (KratosHome) [inline src/components/client/slider/product.tsx:12]: delete comments

File: src/components/client/slider/productSlider.css [hunk 10]
```diff
@@ -0,0 +1,200 @@
+@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;600;700&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Rubik+Doodle+Shadow&display=swap');
```
Comment (KratosHome) [inline src/components/client/slider/productSlider.css:2]: use tailwind

File: src/components/client/slider/productSlider.tsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+import '@splidejs/react-splide/css'
+
+type SplideSliderProps = {
+  products: IProduct[]
+}
+
+export default function ProductSlider({ products }: SplideSliderProps) {
+  return (
+    <div className="mx-auto mb-12 mt-10 max-w-6xl pl-4 pr-0 md:pr-4">
+      {/* Заголовок */}
```
Comment (KratosHome) [inline src/components/client/slider/productSlider.tsx:14]: delete coments

File: src/components/client/slider/productSlider.tsx [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+        <h2 className="rubik-doodle-shadow-regular mb-2 text-[32px] sm:mb-4 sm:text-5xl md:mb-6 md:text-6xl lg:mb-8 lg:text-7xl xl:mb-12 xl:text-8xl">
+          ТОПОВІ СУШЕНИКИ
+        </h2>
+        <div className="swiper-label relative grid place-content-start text-base sm:place-content-end md:text-lg lg:text-xl xl:text-2xl">
+          найсмачніші кусь-топчики
+        </div>
+      </div>
+      <Splide
+        options={{
+          type: 'loop',
```
Comment (KratosHome) [inline src/components/client/slider/productSlider.tsx:25]: move options higher return

File: src/data/products-list.ts [hunk 13]
```diff
@@ -0,0 +1,79 @@
+import { IProduct } from '@/types/IProduct'
+
+export const productsList: IProduct[] = [
```
Comment (KratosHome) [inline src/data/products-list.ts:3]: you have the wrong list of products the most simple example in types there is no title there is a name and so with the rest of the fields

File: src/types/IProduct.d.ts [hunk 14]
```diff
@@ -0,0 +1,9 @@
+export interface IProduct {
+  id: number
```
Comment (KratosHome) [inline src/types/IProduct.d.ts:2]: use the types that already have for products

File: src/components/client/slider/product.tsx [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+                {product.oldPrice} грн
+              </p>
+            )}
+            <p className="text-sm font-semibold sm:text-base lg:text-lg xl:text-xl">
+              {product.price} грн
+            </p>
+          </div>
+
+          {/* Add to Cart Button */}
+          <Button type="button" variant="button">
```
Comment (KratosHome) [inline src/components/client/slider/product.tsx:66]: make one button for the admin and the client, remove the duplicate button

File: src/components/client/slider/product.tsx [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+import { IProduct } from '@/types/IProduct'
+import Button from '@/components/client/ui/button'
+
+type ProductProps = {
+  product: IProduct
+}
+
+const Product = ({ product }: ProductProps) => {
+  return (
+    <div className="max-w-88 relative flex h-full w-full flex-col justify-between gap-[50px] p-2 transition-transform hover:scale-105 hover:shadow-2xl">
```
Comment (KratosHome) [inline src/components/client/slider/product.tsx:11]: weak animation

**Review by KratosHome — APPROVED** [review]

File: src/components/client/slider/product.tsx (hunk 9 above)
Comment (Wasileuss) [reply src/components/client/slider/product.tsx:12]: Done

File: src/components/client/slider/productSlider.tsx (hunk 11 above)
Comment (Wasileuss) [reply src/components/client/slider/productSlider.tsx:14]: Done

File: src/components/client/slider/productSlider.tsx (hunk 12 above)
Comment (Wasileuss) [reply src/components/client/slider/productSlider.tsx:25]: Done

File: src/data/products-list.ts (hunk 13 above)
Comment (Wasileuss) [reply src/data/products-list.ts:3]: Done

File: src/types/IProduct.d.ts (hunk 14 above)
Comment (Wasileuss) [reply src/types/IProduct.d.ts:2]: Done

File: src/app/(app)/globals.css [hunk 17]
```diff
@@ -81,3 +78,6 @@ body {
     }
 }
 
+body {
+  font-family: var(--fontFamily);
```
[the remark this answers is not in the document]
Comment (Wasileuss) [reply src/app/(app)/globals.css:82]: Done

File: src/components/client/slider/product.tsx (hunk 9 above)
Comment (Wasileuss) [reply src/components/client/slider/product.tsx:12]: Done

File: src/components/client/slider/productSlider.tsx (hunk 11 above)
Comment (Wasileuss) [reply src/components/client/slider/productSlider.tsx:14]: Done

File: src/components/client/slider/productSlider.tsx (hunk 12 above)
Comment (Wasileuss) [reply src/components/client/slider/productSlider.tsx:25]: Done

File: src/components/client/slider/product.tsx (hunk 15 above)
Comment (Wasileuss) [reply src/components/client/slider/product.tsx:66]: Done

File: src/components/client/slider/product.tsx (hunk 16 above)
Comment (Wasileuss) [reply src/components/client/slider/product.tsx]: Improved a little bit

File: src/components/client/slider/product.tsx (hunk 15 above)
Comment (Wasileuss) [reply src/components/client/slider/product.tsx]: After removing one button, the build breaks down. I saved it

File: src/types/product.d.ts [hunk 18]
```diff
@@ -10,6 +10,11 @@ interface IProduct {
   statusLabel: string[]
   visited?: number
   nutritionalValue: INutritionalValue
+  available: boolean
```
Comment (KratosHome) [inline src/types/product.d.ts:13]: use existing types

File: src/app/(app)/[locale]/(client)/page.tsx [hunk 19]
```diff
@@ -1,9 +1,12 @@
+'use client'
 import { Hero } from '@/components/client/hero/hero'
+import ProductSlider from '@/components/client/slider/productSlider'
 
 export default function Home() {
   return (
     <main>
       <Hero />
+      <ProductSlider />
```
Comment (KratosHome) [inline src/app/(app)/[locale]/(client)/page.tsx:9]: get the product here and put on the props

File: src/app/(app)/[locale]/admin/products/page.tsx [hunk 20]
```diff
@@ -3,6 +3,7 @@ import React from 'react'
 import { getProducts } from '@/server/products/get-products.server'
 import { ProductList } from '@/components/admin/product-list/product-list'
 import { findProductInfoItems } from '@/server/products/find-product-info-items.server'
+import { IGetProduct, IRecommendations } from '@/types/product'
```
Comment (KratosHome) [inline src/app/(app)/[locale]/admin/products/page.tsx:6]: when wy use d.ts we have auto import

File: src/app/(app)/globals.css [hunk 21]
```diff
@@ -2,82 +2,81 @@
 @tailwind components;
 @tailwind utilities;
 
+@import url('../../components/client/slider/productSlider.css');
```
Comment (KratosHome) [inline src/app/(app)/globals.css:5]: better add import in component where it use

File: src/app/(app)/globals.css [hunk 22]
```diff
@@ -2,82 +2,81 @@
...
+
 body {
-    margin: 0;
-    background-color: white;
-    color: #0f0f0f;
+  @apply m-0 bg-[#fbfbfb] text-[#0f0f0f] dark:bg-[#0f0f0f] dark:text-[#fbfbfb];
```
Comment (KratosHome) [inline src/app/(app)/globals.css:8]: you delete     margin: 0;
background-color: white;
color: #0f0f0f; why?

File: src/components/admin/editor-product/editor-product.tsx [hunk 23]
```diff
@@ -22,6 +22,7 @@ import { editProduct } from '@/server/products/edit-product.server'
 import { useRouter } from 'next/navigation'
 import { convertToBase64 } from '@/utils/convertToBase64'
 import Image from 'next/image'
+import { IProductLocal, IRecommendations } from '@/types/product'
```
Comment (KratosHome) [inline src/components/admin/editor-product/editor-product.tsx:25]: when wy use d.ts we have auto import

File: src/components/admin/product-list/product-list.tsx [hunk 24]
```diff
@@ -35,6 +35,12 @@ import {
 import Image from 'next/image'
 import { FC } from 'react'
 import EditorProduct from '@/components/admin/editor-product/editor-product'
+import {
```
Comment (KratosHome) [inline src/components/admin/product-list/product-list.tsx:38]: when wy use d.ts we have auto import

File: src/components/client/product/product.tsx [hunk 25]
```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react'
```
Comment (KratosHome) [inline src/components/client/product/product.tsx:1]: it client component - add lable

File: src/components/client/product/product.tsx [hunk 26]
```diff
@@ -0,0 +1,130 @@
...
+const Product: React.FC<ProductProps> = ({ product }) => {
+  const locale = useLocale()
+  const [selectedWeightIndex, setSelectedWeightIndex] = useState(0)
+  const selectedPrice = product.variables[selectedWeightIndex].price
+  const selectedNewPrice = product?.variables[selectedWeightIndex].newPrice ?? 0
+  const productName: string =
+    product.name?.[locale as keyof typeof product.name] || 'Назва недоступна'
+
+  return (
+    <CardContainer className="inter-var h-full w-full rounded-sm">
```
Comment (KratosHome) [inline src/components/client/product/product.tsx:29]: use tag link to product

File: src/components/client/product/product.tsx [hunk 27]
```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react'
+import { IProductLocal, IVariableProduct } from '@/types/product'
+import Image from 'next/image'
+import Button from '@/components/client/ui/button'
+import {
+  Select,
```
Comment (KratosHome) [inline src/components/client/product/product.tsx:6]: when wy use d.ts we have auto import

File: src/components/client/product/product.tsx [hunk 28]
```diff
@@ -0,0 +1,130 @@
...
+            ) : (
+              selectedPrice && (
+                <p className="text-sm font-semibold sm:text-base lg:text-lg xl:text-xl">
+                  {selectedPrice} грн
+                </p>
+              )
+            )}
+          </CardItem>
+          <CardItem translateZ={40}>
+            <Button type="button" variant="button">
```
Comment (KratosHome) [inline src/components/client/product/product.tsx:120]: create function who push selected product to console log,     id
name
count
price
newPrise

File: src/components/client/product/product.tsx [hunk 29]
```diff
@@ -0,0 +1,130 @@
...
+            alt={productName}
+            width={229}
+            height={229}
+            className="relative aspect-square h-full w-full object-contain"
+          />
+        </CardItem>
+        <div>Visited: {product.visited}</div>
+        <div className="flex w-full items-center justify-between gap-2">
+          <CardItem translateZ={50}>
+            <h3 className="text-xs font-medium uppercase sm:text-sm md:text-base lg:text-lg xl:text-xl">
```
Comment (KratosHome) [inline src/components/client/product/product.tsx:72]: i think it must bi span teg

File: src/components/client/slider/productSlider.tsx [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+      pagination: 'splide__pagination product__pagination',
+      page: 'splide__pagination__page product__pagination-page',
+    },
+  }
+
+  const [products, setProducts] = useState<IProductLocal[]>([])
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (KratosHome) [inline src/components/client/slider/productSlider.tsx:65]: get product at server

File: src/server/products/create-product.server.ts [hunk 31]
```diff
@@ -2,6 +2,7 @@
 import { connectToDb } from '@/server/connectToDb'
 import { Product } from '@/server/products/productSchema'
 import cloudinary from '@/server/cloudinaryConfig'
+import { IProductLocal } from '@/types/product'
```
Comment (KratosHome) [inline src/server/products/create-product.server.ts:5]: same think when auto import with d.ts

File: src/server/products/get-products.server.ts [hunk 32]
```diff
@@ -1,5 +1,6 @@
 import { connectToDb } from '@/server/connectToDb'
 import { Product } from '@/server/products/productSchema'
+import { IProduct, IProductLocal } from '@/types/product'
```
Comment (KratosHome) [inline src/server/products/get-products.server.ts:3]: same think when auto import with d.ts

File: src/server/products/get-products.server.ts [hunk 33]
```diff
@@ -59,6 +60,12 @@ export async function getProducts(
       variables: product.variables,
       statusLabel: product.statusLabel,
       nutritionalValue: product.nutritionalValue,
+      available: product.available,
```
Comment (KratosHome) [inline src/server/products/get-products.server.ts:63]: we talked about using existing fields

File: src/server/products/get-productsSliderMain.server.ts [hunk 34]
```diff
@@ -0,0 +1,37 @@
+'use server'
+import { connectToDb } from '@/server/connectToDb'
+import { Product } from '@/server/products/productSchema'
+import { IProductLocal, IVariableProduct } from '@/types/product'
+
+export const getProductsSliderMain = async (): Promise<IProductLocal[]> => {
+  console.time('fetch-products')
```
Comment (KratosHome) [inline src/server/products/get-productsSliderMain.server.ts:7]: delete it

File: src/server/products/get-productsSliderMain.server.ts [hunk 35]
```diff
@@ -0,0 +1,37 @@
...
+    await connectToDb()
+
+    const products = (
+      await Product.find()
+        .sort({ visited: -1 })
+        .limit(7)
+        .lean<IProductLocal[]>()
+    ).map((product) => ({
+      ...product,
+      name: product.name,
```
Comment (KratosHome) [inline src/server/products/get-productsSliderMain.server.ts:19]: what card is used as far as I know you get the correct array 
if you have sam error you can say it

File: src/server/products/get-productsSliderMain.server.ts [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+      visited: product.visited,
+      variables: product.variables.map((variant: IVariableProduct) => ({
+        ...variant,
+        weight: Number(variant.weight),
+        price: Number(variant.price),
+        newPrice: variant.newPrice ? Number(variant.newPrice) : 0,
+      })),
+    }))
+
+    console.timeEnd('fetch-products')
```
Comment (KratosHome) [inline src/server/products/get-productsSliderMain.server.ts:31]: delete it

File: src/server/products/get-productsSliderMain.server.ts [hunk 37]
```diff
@@ -0,0 +1,37 @@
...
+    return products
+  } catch (error) {
+    console.error('Failed to fetch products:', error)
+    throw new Error('Failed to fetch top products. Please try again later.')
```
Comment (KratosHome) [inline src/server/products/get-productsSliderMain.server.ts:35]: create better error ansser like enather

File: src/server/products/get-productsSliderMain.server.ts [hunk 38]
```diff
@@ -0,0 +1,37 @@
...
+      variables: product.variables.map((variant: IVariableProduct) => ({
+        ...variant,
+        weight: Number(variant.weight),
+        price: Number(variant.price),
+        newPrice: variant.newPrice ? Number(variant.newPrice) : 0,
+      })),
+    }))
+
+    console.timeEnd('fetch-products')
+    return products
```
Comment (KratosHome) [inline src/server/products/get-productsSliderMain.server.ts:32]: add better answer like another

**Review by KratosHome — APPROVED** [review]

**Review by KratosHome — APPROVED** [review]

File: src/server/products/get-products.server.ts (hunk 33 above)
Comment (Wasileuss) [reply src/server/products/get-products.server.ts:63]: This is no longer used. I just forgot to delete it.

File: src/app/(app)/globals.css (hunk 21 above)
Comment (Wasileuss) [reply src/app/(app)/globals.css:5]: Done

File: src/app/(app)/globals.css (hunk 22 above)
Comment (Wasileuss) [reply src/app/(app)/globals.css:8]: Fixed

File: src/components/client/product/product.tsx (hunk 25 above)
Comment (Wasileuss) [reply src/components/client/product/product.tsx:1]: Done

File: src/components/client/product/product.tsx (hunk 29 above)
Comment (Wasileuss) [reply src/components/client/product/product.tsx:72]: This is not by design. I made a display of the number of visitors for the test.

File: src/server/products/get-productsSliderMain.server.ts (hunk 36 above)
Comment (Wasileuss) [reply src/server/products/get-productsSliderMain.server.ts:31]: Deleted

File: src/server/products/get-productsSliderMain.server.ts (hunk 34 above)
Comment (Wasileuss) [reply src/server/products/get-productsSliderMain.server.ts:7]: Deleted

File: src/app/(app)/[locale]/admin/products/page.tsx (hunk 20 above)
Comment (Wasileuss) [reply src/app/(app)/[locale]/admin/products/page.tsx:6]: Deleted this row

File: src/components/admin/editor-product/editor-product.tsx (hunk 23 above)
Comment (Wasileuss) [reply src/components/admin/editor-product/editor-product.tsx:25]: Deleted this row

File: src/components/client/product/product.tsx (hunk 27 above)
Comment (Wasileuss) [reply src/components/client/product/product.tsx:6]: Deleted this row

File: src/server/products/create-product.server.ts (hunk 31 above)
Comment (Wasileuss) [reply src/server/products/create-product.server.ts:5]: Deleted this row

File: src/server/products/get-productsSliderMain.server.ts (hunk 38 above)
Comment (Wasileuss) [reply src/server/products/get-productsSliderMain.server.ts:32]: ?
I deleted row "console.timeEnd('fetch-products')"

File: src/server/products/get-productsSliderMain.server.ts (hunk 37 above)
Comment (Wasileuss) [reply src/server/products/get-productsSliderMain.server.ts:35]: Done

File: src/server/products/get-productsSliderMain.server.ts (hunk 35 above)
Comment (Wasileuss) [reply src/server/products/get-productsSliderMain.server.ts:19]: Fixed
or I misunderstood

File: src/components/client/slider/productSlider.tsx [hunk 39]
```diff
@@ -0,0 +1,104 @@
...
+      }
+    }
+    fetchData().then((r) => r)
+  }, [])
+
+  if (error) return <div>{error}</div>
+  if (!products.length) return <div>Завантаження...</div>
+
+  return (
+    <div className="mx-auto mb-12 mt-10 max-w-6xl px-0 lg:px-4">
```
[the remark this answers is not in the document]
Comment (Wasileuss) [reply src/components/client/slider/productSlider.tsx:80]: Added padding

File: src/server/products/get-products.server.ts (hunk 32 above)
Comment (Wasileuss) [reply src/server/products/get-products.server.ts:3]: Deleted this row

File: src/components/client/product/product.tsx (hunk 26 above)
Comment (Wasileuss) [reply src/components/client/product/product.tsx:29]: Added

File: src/components/client/product/product.tsx (hunk 28 above)
Comment (Wasileuss) [reply src/components/client/product/product.tsx:120]: Created

**Pull request merged**